### PR TITLE
Add event_name field to `LogRecord`

### PR DESCRIPTION
**Pull request by bIgBV**
Fixes #
Design discussion issue (if applicable) #1349 

## Changes

Add `event_name` field to the `LogRecord` struct.

The `event_name` field is manually passed around through attributes to exporters which then have to parse it out. This change would being it in line with existing logging and tracing frameworks in the rust ecosystem, and provide a performance boost to downstream consumers who depend on the `event_name` field.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Comment by bIgBV** [comment]
I wanted to get this change in first before modifying consumers.

File: opentelemetry/src/logs/record.rs [hunk 1]
```diff
@@ -9,6 +9,9 @@ use std::{borrow::Cow, collections::HashMap, time::SystemTime};
 /// LogRecord represents all data carried by a log record, and
 /// is provided to `LogExporter`s as input.
 pub struct LogRecord {
+    /// Event name. Optional as not all the logging API support it.
```
Comment (cijothomas) [inline opentelemetry/src/logs/record.rs:12]: maybe give a concrete example here of tokio-tracing which do have it.

**Review by cijothomas — APPROVED** [review]
I like this change of treating eventname as a top level field in LogRecord. 
How exporters export this is TBD and can be sorted out in subsequent PRs.

**Review by lalitb — APPROVED** [review]
Thanks for the PR. Looks good. The tracing appended needs changes to populate this field, which can be done as a separate PR.

**Comment by cijothomas** [comment]
Keeping the PR for 1-2 days to see if there are any concerns with this, before merging.

**Pull request merged**